middle_end/flambda/simplify/simplify_set_of_closures.ml

```diff
@@ -368,16 +368,33 @@ let simplify_function context ~used_closure_vars ~shareable_constants
   let params_and_body =
     Code.params_and_body_must_be_present code ~error_context:"Simplifying"
   in
+  let inlining_arguments_from_denv =
+    C.dacc_prior_to_sets context
+    |> DA.denv
+    |> DE.inlining_arguments
+  in
+  (* Compute the set of inlining_arguments used to define this function
+    by taking the "least powerfull" set between the one set in the environment
+    and the one used to define the symbol previously. This way, functions that
+    were imported from a foreign compilation unit after inlining will still
+    considered with a set of inlining arguments coherent with the one used
+    to compile the current file when inlining.
+  *)
```

Comment:
Couple of language nitpicks: typo should be “least powerful”, and the word _be_ is missing before _considered_ (“... will still be considered with ...”)